pan3d/dataset_viewer.py

```diff
@@ -67,13 +66,12 @@ def __init__(
         if state:
             self.state.update(state)
 
-        self._pangeo = pangeo
-        self._esgf = esgf
-        if pangeo:
+        print(catalogs)
+        if "pangeo" in catalogs:
             from pan3d.pangeo_forge import get_catalog
 
             self.state.available_catalogs.append(get_catalog())
-        if esgf:
+        if "esgf" in catalogs:
             from pan3d.esgf import get_catalog
 
             self.state.available_catalogs.append(get_catalog())
```

Comment:
not sure but it would be nice if you could just do it like that

`self.state.available_catalogs = [get_catalog(pkg_name) for pkg_name in catalogs]`

Reply:
Side note, `get_catalog(str)` should not have a if in it...

Reply:
I don't entirely understand this suggestion. I think a list comprehension would look nicer, but I'm not sure how to accomplish the branching import logic here.

Reply:
https://docs.python.org/3.8/library/importlib.html

Reply:
probably not fully correct but something like
```python
def get_catalog(pkg_name):
    try:
        fn = __import__(f"pan3d.{pkg_name}.get_catalog")
        return fn()
   except ImportError:
       raise(....)
```

Reply:
Thanks for clarifying. I was able to do this by using `importlib.import_module` (just using `__import__` gave me errors). Then, I realized that this could be expanded to replace all the places I used `if` branches for catalog function imports. I ended up adding a function `_call_catalog_function` to the `DatasetBuilder` class in [0d37c51](https://github.com/Kitware/pan3d/pull/67/commits/0d37c5161ba190afdd20a7231e70ae105c758d83). Do you think this is a good approach?

Reply:
yes looks much better, thx